scripts/test.sh

```diff
@@ -97,33 +99,31 @@ for i in "${ignored_packages[@]}"; do
   serial_packages=(${serial_packages[@]//*$i*})
 done
 
-if [ "${1:-""}" = "" ]; then
+if [[ -z "${restrict_testing_to}" ]]; then
   for dir in "${packages[@]}"; do
     pushd "$dir"
       ginkgo -p --race -randomizeAllSpecs -randomizeSuites \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        ${@:2}
+        -ldflags="-extldflags=-Wl,--allow-multiple-definition"
     popd
   done
   for dir in "${serial_packages[@]}"; do
     pushd "$dir"
       ginkgo --nodes "${SERIAL_NODES}" --race -randomizeAllSpecs -randomizeSuites -failFast \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        ${@:2}
+        -ldflags="-extldflags=-Wl,--allow-multiple-definition"
     popd
   done
 else
-  dir="${@: -1}"
+  dir="${restrict_testing_to}"
   dir="${dir#./}"
   for package in "${serial_packages[@]}"; do
     if [[ "${dir##$package}" != "${dir}" ]]; then
-      ginkgo --race -randomizeAllSpecs -randomizeSuites -failFast \
-        -ldflags="-extldflags=-Wl,--allow-multiple-definition" \
-        "${@}"
+      pushd "${dir}"
+      ginkgo --nodes "${SERIAL_NODES}" --race -randomizeAllSpecs -randomizeSuites -failFast \
+        -ldflags="-extldflags=-Wl,--allow-multiple-definition"
```

Comment:
Since it exits immediately after the ginkgo and popd wouldn't do anything functional, I didn't bother going down the road of gathering ginkgo's return code in a variable, running popd, and then exiting.